src/unix.rs

```diff
@@ -17,17 +17,19 @@ use std::thread::{self, Builder, JoinHandle};
 use std::time::Duration;
 
 #[derive(Debug)]
-pub enum Client {
-    /// `--jobserver-auth=R,W`
-    Pipe { read: File, write: File },
-    /// `--jobserver-auth=fifo:PATH`
-    Fifo {
-        file: File,
-        path: PathBuf,
-        /// it can only go from false -> true but not the other way around, since that
-        ///  could cause a race condition.
-        is_non_blocking: AtomicBool,
-    },
+enum ClientCreationArg {
+    Fds { read: c_int, write: c_int },
+    Fifo(Box<Path>),
+}
+
+#[derive(Debug)]
+pub struct Client {
+    read: File,
+    write: File,
+    creation_arg: ClientCreationArg,
+    /// it can only go from Some(false) -> Some(true) but not the other way around, since that
+    ///  could cause a race condition.
+    is_non_blocking: Option<AtomicBool>,
```

Comment:
The combination of the name and the `Option` here is not quite obvious. The option indicates whether it **can be** made non-blocking, the boolean indicates whether it **is** nonblocking. The comment could explain that.
It can be understood by reading the code below but having to jump around less should make things easier for new readers.

Reply:
Thanks, updated